### PR TITLE
feat: genesis of the v2 user-first SDK :woohooo:

### PR DESCRIPTION
**Pull request by jsteenb2**
The big changes here are updating the func signatures to remove confusion with error path, APIErrors, and logging. Additionally, we remove much of the start up issues we've had previously, with the logger/config setup, and move that to the new fdk.Run func, which utilizes generics to resolve all the issues we've been seeing pop from authors around the "fn isn't working" notices we've had. All in all, we're trying to reduce the boilerplate, remove bullets from the boostrappping footgun, and provide a more ergonomic experience for the func authors.

### CODE REVIEW

File: sdk.go
Comment (jsteenb2) [inline sdk.go]: One thing I'm still contemplating here..... is if we pass context to the `newHandlerFn`. making it something like:

```
func Run[T Cfg](ctx context.Context, newHandlerFn func(ctx context.Context, cfg T) Handler) {
```

it may make sense, especiallly if anything on the creation of the handler requires some network call/other that would make good on using ctx). WDYAT?

Comment (jsteenb2) [reply sdk.go]: I've convinced myself, going to add the context to the newHandler Fn. Simple change, but makes us robust to nearly any initialization an author has to do.

**Review by johns31459 — DISMISSED** [review]

File: go.mod
Comment (ffalor) [inline go.mod]: hey @jsteenb2 not sure what all is in scope for this, but thoughts on bumping the version of gofalcon that is included? There have been some bugfixes like https://github.com/CrowdStrike/gofalcon/pull/368 & `v0.4.0` introduced many new endpoints and operations from the api spec. There will be a new release this week to include all the changes over the past month which will include some new collections.

Comment (jsteenb2) [reply go.mod]: sure thing, we can get that version upgraded here 👍

Comment (ffalor) [reply go.mod]: Working with John to update the sdk to include the new foundry collections (storage, workflow) it should be out today

Comment (jsteenb2) [reply go.mod]: excellent, I'll wait to bring that in before I merge 👍

Comment (ffalor) [reply go.mod]: Delayed to implement another collection that he wants & he found a few things wrong with the current spec. I'll comment back when this is done.

Comment (jsteenb2) [reply go.mod]: we can always add a patch version to `v0.4.1`, I won't block on this update for now. I have some work to do, but we're more than capable of rolling a new version with small updates anytime 👍

Comment (jsteenb2) [reply go.mod]: @ffalor , any update on the patched version of `gofalcon`?

File: examples/fn_config/main.go
Comment (chriscannon) [inline examples/fn_config/main.go]: Using an `X-CS-*` header here is going to be problematic if the request goes through the API Gateway because it filters out all `X-CS-*` headers.

Comment (jsteenb2) [reply examples/fn_config/main.go]: oiii fair point, I can remove that `X-Cs` prefix and make it `X-Fn-` for the examples

**Review by chriscannon — DISMISSED** [review]

**Review by chriscannon — APPROVED** [review]

**Pull request merged**
